matscipy/dislocation.py

```diff
@@ -2827,148 +2702,251 @@ def displacements(self, bulk_positions, center,
-                                      pbc=(False, False, True),
-                                      latticeconstant=alat)
 
-        disloCenterX = 0.0
-        # middle between two (111) planes
-        disloCenterY = unit_cell.cell[1][1] / 6.0
-
-        unit_cell_core_position = np.array([disloCenterX,
-                                            disloCenterY, 0])
+class FCCScrew110Dislocation(CubicCrystalDissociatedDislocation):
+    burgers_dimensionless = np.array([1, -1, 0]) / 2
+    new_left_burgers = np.array([2., -1., -1.]) / 6
+    left_dislocation = FCCScrewShockleyPartial
+    right_dislocation = FCCScrewShockleyPartial
 
-        glide_distance = alat * np.linalg.norm(axes[0]) / 4.0
 
-        n_planes = 6
-        self_consistent = True
-        super().__init__(unit_cell, alat, C11, C12, C44,
-                         axes, burgers, unit_cell_core_position, parity,
-                         glide_distance, n_planes=n_planes,
-                         self_consistent=self_consistent)
+class FCCEdge110Dislocation(CubicCrystalDissociatedDislocation):
+    crystalstructure = "fcc"
+    burgers_dimensionless = np.array([1, -1, 0]) / 2
+    new_left_burgers = np.array([2., -1., -1.]) / 6
+    left_dislocation = FCCEdgeShockleyPartial
+    right_dislocation = FCCEdgeShockleyPartial
+
+
+def _validate_disloc_inputs(a, symbol="w", axes=None, crystalstructure="fcc", pbc=[True, True, True]):
+    ''''
+    Validate & transform args of dislocation class __init__ methods
+
+    a: float or ase.atoms
+        EITHER lattice constant (in A), or the cubic bulk structure
+    symbol: str
+        Elemental symbol, passed to relevant crystalstructure generator when a is float
+    axes: np.array
+        Axes transform to apply to bulk cell
+    crystalstructure: str
+        Base Structure of bulk system, currently supported: fcc, bcc, diamond
+    pbc: list of bool
+        Periodic Boundary Conditions in x, y, z
+    '''
 
+    from ase.lattice.cubic import FaceCenteredCubic, BodyCenteredCubic, Diamond
+    from ase.atoms import Atoms
+    from ase.build import cut, rotate
+    from warnings import warn
+
+    constructors = {
+        "fcc": FaceCenteredCubic,
+        "bcc": BodyCenteredCubic,
+        "diamond": Diamond
+    }
+
+    # Choose correct ase.lattice.cubic constructor given crystalstructure
+    try:
+        cell_builder = constructors[crystalstructure.lower()]
+    except (KeyError, AttributeError) as e:
+        if type(e) == KeyError:
+            # KeyError when crystalstructure not a key in constructors
+            raise ValueError(f"Crystal Structure {crystalstructure} not in accepted list: {list(constructors.keys())}")
+        else:
+            # AttributeError when type(crystalstructure) doesn't support .lower() (not a valid input type)
+            raise TypeError(f"crystalstructure argument {crystalstructure} is not a string")
+
+
+    if np.issubdtype(type(a), np.floating) or np.issubdtype(type(a), np.integer):
```

Comment:
also check array size=0 (i.e. it's a scalar). Check this but I think `np.array(alat).size == 0` when alat is an int or float scalar.